Zpp/Simulation/Simulator.cs

```diff
@@ -0,0 +1,130 @@
+
+            _currentTime = simulationInterval.StartAt;
+            _akkaSimulation = new AkkaSim.Simulation(_simulationConfig);
+            var jobDistributor = _akkaSimulation.ActorSystem
+                                                .ActorOf(props: JobDistributor.Props(_akkaSimulation.SimulationContext, _currentTime)
+                                                        , name: "JobDistributor");
+
+            // ToDo reflect CurrentTimespawn ?
+            _akkaSimulation.Shutdown(simulationInterval.EndAt);
+            // Create a Machines
+            CreateResource(jobDistributor);
+            
+            // Set purchased Demands finished.
+            ProvideRequiredPurchaseForThisInterval(simulationInterval);
+            
+            // Distribute Ready Jobs
+            ProvideJobDistributor(simulationInterval, jobDistributor);
+
+            // TODO What to do with finished Jobs? How is PrO connected to the StockExchange. 
+            /// a. Provide a Stockexchange Key with every ProductionOrder to complete SE.
+            /// b. delete all Stockexchanges that are "ToProduce"
+            /// --> _c. satisfy the first not yet satisfied Stockexchange 
+            // Handle JobFinish
+
+            var monitor = _akkaSimulation.ActorSystem
+                                         .ActorOf(props: WorkTimeMonitor.Props(time: _currentTime),
+                                                   name: "SimulationMonitor");
+            if(_akkaSimulation.IsReady())
+            {
+                _akkaSimulation.RunAsync();
+                Continuation(_simulationConfig.Inbox, _akkaSimulation);
+            }
+
+
+
+            Debug.WriteLine("System shutdown. . . ");
+            Debug.WriteLine("System Runtime " + _akkaSimulation.ActorSystem.Uptime);
+            return true;
+        }
+
+        private void ProvideJobDistributor(SimulationInterval simulationInterval, IActorRef jobDistributor)
+        {
+            var operationManager = new OperationManager(_dbMasterDataCache, _dbTransactionData);
+            _akkaSimulation.SimulationContext
+                           .Tell(JobDistributor.OperationsToDistibute
+                                                     .Create(operationManager, jobDistributor)
+                                                            , ActorRefs.NoSender);
+        }
+
+        /// <summary>
+        /// can be done by some sort of StockManager later if time periods not static
+        /// </summary>
+        /// <param name="simulationInterval"></param>
+        private void ProvideRequiredPurchaseForThisInterval(SimulationInterval simulationInterval)
+        {
+            var from = new DueTime((int)simulationInterval.StartAt);
+            var to = new DueTime((int)simulationInterval.EndAt);
+            var stockExchanges = _dbTransactionData.GetAggregator().GetProvidersForInterval(from, to); 
+                // .GetAll StockExchangeProvidersGetAll().GetAll();
```

Comment:
Usually, this should be avoided. Code is active or is deleted. Commenting out code is in times of version control and code snippets around the web not really needed anymore.

Reply:
Deleted.